tests/general/CMakeLists.txt

```diff
@@ -348,20 +348,20 @@ else ()
 endif ()
 
 #===== pio_iosystems_test2 =====
-# add_executable (pio_iosystem_tests2 EXCLUDE_FROM_ALL
-#     pio_iosystem_tests2.F90
-#     ${CMAKE_CURRENT_SOURCE_DIR}/util/pio_tutil.F90)
-# target_link_libraries (pio_iosystem_tests2 piof)
-# add_dependencies (tests pio_iosystem_tests2)
+add_executable (pio_iosystem_tests2 EXCLUDE_FROM_ALL
+    pio_iosystem_tests2.F90
+    ${CMAKE_CURRENT_SOURCE_DIR}/util/pio_tutil.F90)
+target_link_libraries (pio_iosystem_tests2 piof)
+add_dependencies (tests pio_iosystem_tests2)
 
-# if (PIO_USE_MPISERIAL)
-#     add_test(NAME pio_iosystem_tests2
-#         COMMAND pio_iosystem_tests2)
-#     set_tests_properties(pio_iosystem_tests2
-#         PROPERTIES TIMEOUT ${DEFAULT_TEST_TIMEOUT})
-# else ()
-#     add_mpi_test(pio_iosystem_tests2
-#         EXECUTABLE ${CMAKE_CURRENT_BINARY_DIR}/pio_iosystem_tests2
-#         NUMPROCS 4
-#         TIMEOUT ${DEFAULT_TEST_TIMEOUT})
-# endif ()
+if (PIO_USE_MPISERIAL)
+    add_test(NAME pio_iosystem_tests2
+        COMMAND pio_iosystem_tests2)
+    set_tests_properties(pio_iosystem_tests2
+        PROPERTIES TIMEOUT ${DEFAULT_TEST_TIMEOUT})
+else ()
+    # add_mpi_test(pio_iosystem_tests2
```

Comment:
When you check in tests that fail, it breaks my continuous integration testing.

(That is, I have a jenkins task that is building the code every time it changes, so that I can see if I break something. If you check in a test that fails, then it shows as broken every time, defeating the purpose of continuous integration.)

At the moment let's just leave this as it is until I get my current PR merged. Then I would like to turn off the test on master again, so that I can make the changes that we discussed WRT fully merging in the async code so that there are no longer two different versions of some PIO functions.

But the test will still be there, and as soon as I am done, you can take a branch, turn the test back on in the branch, and proceed with your fix.